lib/rspec/matchers/dsl.rb

```diff
@@ -29,7 +29,7 @@ module DSL
       # @see RSpec::Matchers
       def alias_matcher(new_name, old_name, options={}, &description_override)
         description_override ||= lambda do |old_desc|
-          old_desc.gsub(EnglishPhrasing.split_words(old_name), EnglishPhrasing.split_words(new_name))
+          old_desc.sub(EnglishPhrasing.split_words(old_name), EnglishPhrasing.split_words(new_name))
```

Comment:
I think this needs to only match the start of the string, this might still accidentally override terms if the original description is not generated.

Reply:
Thanks for the feedback! Can you clarify this a bit? `#sub` matches the first occurrence, so are you saying something like matching the first n characters of the string?

Also, what do you mean by if the original description is not generated? Since `BaseMatcher` implements description, I would think that they all would have a description. Unless a class implements its own description which does not contain the name of the matcher.

Reply:
> Can you clarify this a bit? #sub matches the first occurrence, so are you saying something like matching the first n characters of the string?

`"hello_old_name".sub("old_name","new_name")` would produce `"hello_new_name"` but we don't want that. We only want to replace `old_name` as it was generated initially. 

> Also, what do you mean by if the original description is not generated? Since BaseMatcher implements description, I would think that they all would have a description. 

Similarly it is possible for matchers to implement `description` themselves and it might mean they don't contain the old_name at all, in this case we shouldn't be replacing the first instance blindly as it might not be in the description but in the expected.